scripts/jenkins/cloud/manual/copy_clouds.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python
+
+import argparse
+import os
+
+import yaml
+
+
+def parse_commandline():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "SRC",
+        help="The source directory or file for the clouds")
+    parser.add_argument(
+        "DEST",
+        help="The full path of the finaly clouds.yaml")
+    parser.add_argument(
+        "--set",
+        metavar="KEY:VALUE",
+        help="Set KEY to VALUE",
+        default=[],
+        action="append")
+    return parser.parse_args()
+
+
+def build_file_list(source):
+    if os.path.isdir(source):
+        return [f.path for f in os.scandir(source)
+                if f.is_file() and f.name.endswith(".yaml")]
+    else:
+        if os.path.isfile(source):
+            return [source]
+        else:
+            raise Exception(
+                "{} is neither file nor directory".format(source))
+
+
+def merge_dictionaires(a, b):
+    if not (isinstance(a, dict) and isinstance(b, dict)):
+        raise Exception("Expected two dictionaries but got {} and {}".format(
+            a.__class__.__name__, b.__class__.__name__))
+
+    result = a.copy()
+    for k, v in b.items():
+        if k in result and isinstance(result[k], dict) and isinstance(v, dict):
+            result[k] = merge_dictionaires(result[k], v)
+        else:
+            result[k] = v
+
+    return result
+
+
+def read_config(files):
+    config = {}
+    for f in files:
+        with open(f) as fd:
+            config = merge_dictionaires(config, yaml.load(
+                fd.read(), Loader=yaml.SafeLoader))
+
+    return config
+
+
+def replace_one_key(config, key, new_value):
+    for k, v in config.items():
+        if k == key:
+            config[k] = new_value
+        elif isinstance(config[k], dict):
+            replace_one_key(config[k], key, new_value)
+
+
+def replace_keys(config, updates):
+    new_config = config.copy()
+    for update in updates:
+        replace_one_key(new_config, *update.split(":"))
+
+    return new_config
+
+
+def main(options):
+    config = replace_keys(read_config(
+        build_file_list(options.SRC)), options.set)
+    os.makedirs(os.path.dirname(options.DEST))
```

Comment:
if the deployer-entry-point.sh is run more than once this fails with: 
Traceback (most recent call last):
  File "/opt/bin/copy_clouds.py", line 87, in <module>
    main(parse_commandline())
  File "/opt/bin/copy_clouds.py", line 81, in main
    os.makedirs(os.path.dirname(options.DEST))
  File "/opt/ansible-venv-python3/lib64/python3.7/os.py", line 221, in makedirs
    mkdir(name, mode)
FileExistsError: [Errno 17] File exists: '/root/.config/openstack'

It could cause other problems if it fails before something is complete, like in the case where not all files have been copied due to an unrelated error. 

you can catch the exception in order to get passed the error, like so:

    try:
        os.makedirs(os.path.dirname(options.DEST))
    except FileExistsError:
        pass